corehq/apps/ota/views.py

```diff
@@ -169,7 +169,14 @@ def phone_holds_all_cases(request):
         except MissingSyncLog:
             return False
 
-    if not case_ids_to_claim and phone_holds_all_cases(request):
+    def cases_to_claim_modified_since_last_synclog_date():
+        try:
+            synclog = get_properly_wrapped_sync_log(request.last_sync_token)
+        except MissingSyncLog:
+            return False
+        return bool(CommCareCase.objects.get_modified_case_ids(domain, list(case_ids_to_claim), synclog))
```

Comment:
Looking at this again I see that we are only checking cases that need to be claimed (i.e. a pre-filtered list). For this to work we need to check ALL the case IDs so this should be changed:
```suggestion
        return bool(CommCareCase.objects.get_modified_case_ids(domain, case_ids, synclog))
```